pkg/sqlite/load.go

```diff
@@ -642,6 +642,13 @@ func (s *sqlLoader) getCSVNames(tx *sql.Tx, packageName string) ([]string, error
 		return nil, err
 	}
 
+	// Check if the query returned zero rows.
+	// In this case, we can make the assumption that
+	// the package we're querying for doesn't exist.
+	if len(csvNames) == 0 {
```

Comment:
hmm, I think we still need to clean up the other tables that represent a package; e.g. `package`. It looks like the `RemovePackage` method below would return before cleaning up the other table(s).

Reply:
Converting this to a warning message may be the most straightforward solution to allow existing cleanup functions to run in `RemovePackage` and it still improves the UX. I'm still trying to wrap my head around what situations we would care about cleanup when attempting to removing a package that doesn't exist.

Reply:
The errors for each package from the list of packages is aggregated in the calling site, so I'm not sure going with either an error or warn really changes things. Either should work I think. 
  
However, I don't think `getCSVNames` guarantees that if no CSV exists for a package, the package itself is non-existent,  which is what I think @njhale's point is too. 
Maybe using the result of the [Exec](https://github.com/operator-framework/operator-registry/blob/master/pkg/sqlite/load.go#L673) statement to see if `RowsAffected` is 0 is a better time to throw the error/warning.

Reply:
I ended up moving the logic of checking if a package exists into its own method, but I'm fine with deflating that logic back into `getCSVNames` if we don't think there's value going forward in maintaining that method. That method simply queries the `package` table for whether that `@packageName` parameter exists in the name column, and if it doesn't (i.e. no rows were returned by the `db.Query(...)` then return an error. I [made a quick review comment](https://github.com/operator-framework/operator-registry/pull/545#discussion_r554469440) on whether that table should serve as the source of truth for package existence.

Updated the `RemovePackage` to avoid exiting early if we encounter a package that doesn't exist in the registry database in order to ensure that the other tables that may be tracking bundles or channels associated with that package can have a chance to be cleaned up just in the case.

Reply:
I also wasn't aware of that `utilerrors.Aggregate` package until I dived into the registry code, but at a glance, that package is definitely a nice interface for handling errors and much cleaner than just appending errors and returned that joined error string using something `fmt.Errorf(...)`.

Reply:
FYI my thoughts on `checkIfPackageExists` are in [this comment](https://github.com/operator-framework/operator-registry/pull/545#discussion_r555039538).